### ISSUE TITLE
Multiple config files gives bogus netsnmp deprecation warning if agent section isn't in first file read 

### ISSUE THREAD

**Issue by poodad** [influxdata/telegraf#12454]
### Relevant telegraf.conf

```toml
#main.conf
[agent]
 snmp_translator = "gosmi"

#inputs-windows.conf

[[inputs.win_perf_counters.object]]
  ObjectName = "PhysicalDisk"
  Instances = ["*"]
  Counters = ["Avg. disk sec/read","Avg. disk sec/write"]
  Measurement = "win_disk"
```

### Logs from Telegraf

```text
2023-01-03T21:19:42Z W! DeprecationWarning: Value "netsnmp" for option "snmp_translator" of plugin "agent" deprecated since version 1.25.0 and will be removed in 2.0.0: Use 'gosmi' instead
2023-01-03T21:19:42Z W! DeprecationWarning: Value "netsnmp" for option "snmp_translator" of plugin "agent" deprecated since version 1.25.0 and will be removed in 2.0.0: Use 'gosmi' instead
```

### System info

telegraf 1.25.0 on Windows

### Steps to reproduce

1. Create config file main.conf - populate with [agent] section stuff including snmp_translator = "gosmi" and outputs section
2. Create other config files named "inputs-whatever" with input sections
3. run telegraf in --test mode
...

### Expected behavior

expect telegraf to read all config files in --config-directory before complaining about deprecated features.

### Actual behavior

Telegraf complains about deprecated snmp_translator for every config file read until the file with [agent] snmp_translator is read.

### Additional info

renaming files such that the file with snmp_translator is read first (ie from main.conf to 00main.conf) is a workaround.

**Comment by powersj** [comment]
> run telegraf in --test mode
> expect telegraf to read all config files in --config-directory before complaining about deprecated features.

Based on your steps to reproduce it sounds like you are using multiple `--config` options, but then your expected behavior mentions the `--config-directory` option. As such this took longer to reproduce than it should have.

I have the following structure:

```
telegraf
config.toml
configs/
└── inputs-windows.conf
```

These all work as expected:

```
./telegraf --debug --once --config main.conf
./telegraf --debug --once --config main.conf --config configs/inputs-windows.conf
./telegraf --debug --once --config configs/inputs-windows.conf --config main.conf 
./telegraf --debug --once --config main.conf --config-directory=configs
./telegraf --debug --once --config-directory=configs --config main.conf 
```

However, once I moved both files into a directory, created an empty other.toml and ran the deprecation messages show up:

```
./telegraf --debug --once --config-directory=configs --config other.toml
```

> expect telegraf to read all config files in --config-directory before complaining about deprecated features.

The intention is that this deprecation is only get printed if a user explicitly states to use "netsnmp". If the value is unset, then this is ignored, and then the translator is set to "netsnmp" since it is still the default. This was getting done for every file that is read and not at the very end of reading files. I have put up a PR #12454 to move this.

The reason you get multiple deprecation warnings is because telegraf reads all the configs multiple times. This is something else we should look into, but in a different issue.

**Comment by poodad** [comment]
Slightly incorrect
Running ./telegraf --debug --once --config main.conf --config-directory=configs
will produce a bogus warning for every file in the "configs" directory that is read before it gets to the file that has the snmp_translate value.

From my limited testing, telegraf appears to read files in --config-directory in alphabetical order, so if you have the following files:
FileA.conf
FileB.conf
FileC.conf
ThisFileHasTheSnmp_TranslateCommandInIt.conf

You will get 3 three bogus warnings. If you delete "FileC.conf" and re-run, you'll get two bogus warnings.

**Comment by powersj** [comment]
> Running ./telegraf --debug --once --config main.conf --config-directory=configs
> will produce a bogus warning for every file in the "configs" directory that is read before it gets to the file that has the snmp_translate value.

It seems master is already different than 1.25:

```
❯ ./telegraf --debug --once --config config.conf --config-directory=configs
2023-01-04T16:51:38Z I! Starting Telegraf 1.26.0-f5b2d300
2023-01-04T16:51:38Z I! Available plugins: 228 inputs, 9 aggregators, 26 processors, 21 parsers, 57 outputs, 2 secret-stores
2023-01-04T16:51:38Z I! Loaded inputs: mem (2x)
2023-01-04T16:51:38Z I! Loaded aggregators: 
2023-01-04T16:51:38Z I! Loaded processors: 
2023-01-04T16:51:38Z I! Loaded secretstores: 
2023-01-04T16:51:38Z I! Loaded outputs: file
2023-01-04T16:51:38Z I! Tags enabled: host=ryzen
2023-01-04T16:51:38Z D! [agent] Initializing plugins
2023-01-04T16:51:38Z D! [agent] Connecting outputs
2023-01-04T16:51:38Z D! [agent] Attempting connection to [outputs.file]
2023-01-04T16:51:38Z D! [agent] Successfully connected to outputs.file
2023-01-04T16:51:38Z D! [agent] Starting service inputs
2023-01-04T16:51:38Z D! [agent] Stopping service inputs
2023-01-04T16:51:38Z D! [agent] Input channel closed
2023-01-04T16:51:38Z I! [agent] Hang on, flushing any cached metrics before shutdown
mem,host=ryzen commit_limit=21077434368i,high_free=0i,low_free=0i,vmalloc_used=139673600i,total=33564942336i,available=22436139008i,cached=14011260928i,free=9337249792i,slab=790958080i,buffered=3858432i,high_total=0i,huge_pages_free=0i,inactive=9234604032i,sreclaimable=460316672i,sunreclaim=330641408i,used_percent=30.426309337187593,dirty=1990656i,huge_pages_total=0i,low_total=0i,mapped=1663762432i,swap_free=4294963200i,vmalloc_total=35184372087808i,used=10212573184i,available_percent=66.84396708745771,huge_page_size=2097152i,page_tables=71458816i,shared=440823808i,committed_as=25282248704i,swap_total=4294963200i,write_back=0i,write_back_tmp=0i,active=13682798592i,swap_cached=0i,vmalloc_chunk=0i 1672851098000000000
mem,host=ryzen available=22436139008i,low_free=0i,slab=790958080i,sreclaimable=460316672i,available_percent=66.84396708745771,active=13682798592i,cached=14011260928i,huge_page_size=2097152i,mapped=1663762432i,vmalloc_used=139673600i,dirty=1990656i,vmalloc_chunk=0i,used_percent=30.426309337187593,free=9337249792i,inactive=9234604032i,vmalloc_total=35184372087808i,write_back_tmp=0i,sunreclaim=330641408i,swap_free=4294963200i,total=33564942336i,used=10212573184i,committed_as=25282248704i,high_free=0i,page_tables=71458816i,shared=440823808i,write_back=0i,buffered=3858432i,commit_limit=21077434368i,swap_total=4294963200i,huge_pages_total=0i,swap_cached=0i,high_total=0i,huge_pages_free=0i,low_total=0i 1672851098000000000
2023-01-04T16:51:38Z D! [outputs.file] Wrote batch of 2 metrics in 29.6µs
2023-01-04T16:51:38Z D! [outputs.file] Buffer fullness: 0 / 10000 metrics
2023-01-04T16:51:38Z I! [agent] Stopping running outputs
2023-01-04T16:51:38Z D! [agent] Stopped Successfully
❯ ../telegraf-builds/telegraf-v1.25.0 --debug --once --config config.conf --config-directory=configs
2023-01-04T16:52:16Z W! DeprecationWarning: Value "netsnmp" for option "snmp_translator" of plugin "agent" deprecated since version 1.25.0 and will be removed in 2.0.0: Use 'gosmi' instead
2023-01-04T16:52:16Z I! Starting Telegraf 1.25.0
2023-01-04T16:52:16Z I! Available plugins: 228 inputs, 9 aggregators, 26 processors, 21 parsers, 57 outputs, 2 secret-stores
2023-01-04T16:52:16Z I! Loaded inputs: mem (2x)
2023-01-04T16:52:16Z I! Loaded aggregators: 
2023-01-04T16:52:16Z I! Loaded processors: 
2023-01-04T16:52:16Z I! Loaded secretstores: 
2023-01-04T16:52:16Z I! Loaded outputs: file
2023-01-04T16:52:16Z I! Tags enabled: host=ryzen
2023-01-04T16:52:16Z D! [agent] Initializing plugins
2023-01-04T16:52:16Z D! [agent] Connecting outputs
2023-01-04T16:52:16Z D! [agent] Attempting connection to [outputs.file]
2023-01-04T16:52:16Z D! [agent] Successfully connected to outputs.file
2023-01-04T16:52:16Z D! [agent] Starting service inputs
2023-01-04T16:52:16Z D! [agent] Stopping service inputs
2023-01-04T16:52:16Z D! [agent] Input channel closed
2023-01-04T16:52:16Z I! [agent] Hang on, flushing any cached metrics before shutdown
mem,host=ryzen used=10227884032i,available_percent=66.81129909747509,committed_as=25307037696i,low_free=0i,low_total=0i,slab=790925312i,vmalloc_used=139747328i,write_back=0i,active=13847109632i,huge_pages_total=0i,vmalloc_total=35184372087808i,inactive=9233186816i,vmalloc_chunk=0i,available=22425174016i,huge_page_size=2097152i,sunreclaim=330493952i,swap_cached=0i,page_tables=71712768i,total=33564942336i,used_percent=30.471924931716945,buffered=3858432i,commit_limit=21077434368i,dirty=909312i,high_total=0i,sreclaimable=460431360i,cached=14160945152i,free=9172254720i,high_free=0i,huge_pages_free=0i,mapped=1661779968i,shared=436031488i,swap_free=4294963200i,swap_total=4294963200i,write_back_tmp=0i 1672851136000000000
mem,host=ryzen cached=14160945152i,mapped=1661779968i,page_tables=71712768i,total=33564942336i,swap_cached=0i,swap_total=4294963200i,vmalloc_used=139747328i,low_total=0i,buffered=3858432i,committed_as=25307037696i,inactive=9233186816i,shared=436031488i,vmalloc_chunk=0i,used=10227884032i,high_total=0i,sunreclaim=330493952i,swap_free=4294963200i,write_back=0i,free=9172254720i,slab=790925312i,huge_pages_free=0i,sreclaimable=460431360i,write_back_tmp=0i,used_percent=30.471924931716945,active=13847109632i,dirty=909312i,huge_page_size=2097152i,huge_pages_total=0i,vmalloc_total=35184372087808i,available_percent=66.81129909747509,commit_limit=21077434368i,high_free=0i,low_free=0i,available=22425174016i 1672851136000000000
2023-01-04T16:52:16Z D! [outputs.file] Wrote batch of 2 metrics in 31.94µs
2023-01-04T16:52:16Z D! [outputs.file] Buffer fullness: 0 / 10000 metrics
2023-01-04T16:52:16Z I! [agent] Stopping running outputs
2023-01-04T16:52:16Z D! [agent] Stopped Successfully

```

In any case my PR should further clean this up.
